Remove hard stop (error) added to conf.c, Solaris can use fcntl().

The Release Notes states:

	NOTE: File locking using fcntl() does not interoperate
	with Berkeley DB 5.x (and probably later). Use
	CDB, flock() (-DHASFLOCK), or an earlier Berkeley
	DB version. Problem noted by Harald Hannelius.

And KNOWBUGS states:

        For Linux the default is to use fcntl() for file locking.  However,
        this does not work with Berkeley DB 5.x and probably later.
        Switching to flock(), i.e., compile with -DHASFLOCK fixes this
        (however, the have been problems with flock() on some Linux
        versions). Alternatively, use CDB or an earlier BDB version.

On Solaris we have not observed issues with fcntl(), and flock() is not an option.
Testing provided in libsmutil/t-lockfile.c / libsmutil/t-maplock-0.sh passes.

Patch to be passed upstream.
diff -r 1fd530d3cc2b sendmail/conf.c
--- a/sendmail/conf.c	Wed Mar 08 05:53:35 2023 -0800
+++ b/sendmail/conf.c	Wed Mar 08 06:26:08 2023 -0800
@@ -5957,7 +5957,7 @@
 #endif
 #if NEWDB
 # if defined(DB_VERSION_MAJOR) && defined(DB_VERSION_MINOR)
-#  if DB_VERSION_MAJOR >= 5 && !HASFLOCK
+#  if DB_VERSION_MAJOR >= 5 && !defined(SOLARIS) && !HASFLOCK
 #    ERROR "Berkeley DB file locking needs flock() for version 5.x (and greater?)"
 #  endif
 	"NEWDB=" SM_XSTR(DB_VERSION_MAJOR) "." SM_XSTR(DB_VERSION_MINOR),
